src/rsa.rs

```diff
@@ -244,6 +255,49 @@ impl RSAKeyPair {
     pub fn public_modulus_len(&self) -> usize {
         unsafe { RSA_size(self.rsa.as_ref()) }
     }
+
+    /// Sign the given message, hashed with the given digest algorithm.
+    pub fn sign(&self, digest_alg: &'static digest::Algorithm, msg: &[u8], signature: &mut [u8]) -> Result<(), ()> {
+        if signature.len() != self.public_modulus_len() {
+            return Err(());
+        }
+
+        let found = PREFIXES.into_iter().find(|pair| pair.0.nid == digest_alg.nid);
+        if found.is_none() {
+            return Err(());
+        }
```

Comment:
There's a problem with doing it this way: Since `PREFIXES` references every digest algorithm, that means that all the code for every digest algorithm has to be linked into the program, even if only a subset are used.

Instead, I suggest that we rename `RSA_PKCS1` to `PKCS1Padding`, make it public, and then have `sign` take `padding: &PKCS1Padding` instead of `digest_alg` as a parameter.